src/main/java/com/google/devtools/build/lib/rules/objc/ObjcVariablesExtension.java

```diff
@@ -189,9 +191,18 @@ private void addArchiveVariables(CcToolchainVariables.Builder builder) {
   private void addFullyLinkArchiveVariables(CcToolchainVariables.Builder builder) {
     builder.addStringVariable(
         FULLY_LINKED_ARCHIVE_PATH_VARIABLE_NAME, fullyLinkArchive.getExecPathString());
+
+    ImmutableSet<Artifact> ccLibs =
+        new ImmutableSet.Builder()
+        .addAll(objcProvider.getCcLibraries())
+        .build();
```

Comment:
```suggestion
    ImmutableSet<Artifact> ccLibs = ImmutableSet.copyOf(objcProvider.getCcLibraries());
```

Reply:
Thanks! I was trying to find the nicer invocation here

Reply:
`copyOf` is most likely more efficient. Depending on the type of `getCcLibraries` it can allocate a set of right size and copy elements from it without resizing. A builder allocates a buffer of default size which is fine for small collections, but with large collections it would thrown away right after `addAll` call.